web/workflow_modular.R

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env Rscript
+
+library("PEcAn.all")
+library(optparse)
+
+#------------------------------------------------------------------------------
+# Command Line Interface
+#------------------------------------------------------------------------------
+option_list <- list(
+    make_option(c("-s", "--settings"), type="character", default=NULL,
+                help="Settings file path", metavar="FILE"),
+    make_option(c("-m", "--module"), type="character", default="all",
+                help="Module to run [all|trait|model]"),
+    make_option(c("--debug"), action="store_true", default=FALSE,
+                help="Run in debug mode")
+)
+
+opt_parser <- OptionParser(option_list=option_list)
+opts <- parse_args(opt_parser)
+
+if (is.null(opts$settings)) {
+    stop("Settings file must be provided. Use --help for usage information.")
+}
+
+# make sure always to call status.end
+options(warn = 1)
+options(error = quote({
+  try(PEcAn.utils::status.end("ERROR"))
+  try(PEcAn.remote::kill.tunnel(settings))
+  if (!interactive()) {
+    q(status = 1)
+  }
+}))
+
+# ----------------------------------------------------------------------
+# Independent Module Functions
+# ----------------------------------------------------------------------
+run_trait_analysis <- function(settings_path, debug = FALSE) {
+  # Load settings
+  settings <- PEcAn.settings::read.settings(settings_path)
+  
+  if (PEcAn.utils::status.check("TRAIT") == 0) {
+    if (debug) cat("Starting trait analysis...\n")
+    
+    PEcAn.utils::status.start("TRAIT")
+    settings <- PEcAn.workflow::runModule.get.trait.data(settings)
+    PEcAn.settings::write.settings(settings, outputfile = "pecan.TRAIT.xml")
+    PEcAn.utils::status.end()
+  }
+  
+  # Run meta-analysis if configured
+  if (!is.null(settings$meta.analysis)) {
+    if (PEcAn.utils::status.check("META") == 0) {
+      if (debug) cat("Starting meta-analysis...\n")
+      
+      PEcAn.utils::status.start("META")
+      PEcAn.MA::runModule.run.meta.analysis(settings)
+      PEcAn.utils::status.end()
+    }
+  }
+  
+  return(settings)
+}
+
+run_model_execution <- function(settings_path, debug = FALSE) {
+  # Load settings
+  settings <- PEcAn.settings::read.settings(settings_path)
```

Comment:
I don't think you want to re-load the settings object each time. Just pass the settings object, not the settings path